src/main/cpp/faultinj/README.md

```diff
@@ -0,0 +1,98 @@
+# Fault Injection Tool for CUDA Runtime and CUDA Driver API
+
+The goal of this tool is to increase testability of the failure handling logic
+in CUDA applications.
+
+It's especially important when CUDA is embedded in a higher-level fault-tolerant framework
+to ensure that CUDA failures are handled correctly in that
+- fatal errors leaving a GPU in unusbale state are detected and such GPUs are prevented from executing retries for
+a failed computation
+- non-fatal errors are retried without losing valuable compute resources
+- error handling logic does not cause deadlocks and other sort of unresponsiveness.
+
+This tool allows creating test scenarios and triggering CUDA errors at will
+with some degree of configurability to determine whether an individual CUDA process
+or the higher-level framework such as Apache Spark remain usable when it should
+or gracefully exits with an actionable error message when the errors are irrecoverable.
+
+## Deployment
+
+The tool is designed with automated testing and interactive testing use cases in mind. The tool is a dynamically linked library `libcufaultinj.so` that is loaded by the CUDA process via CUDA Driver API `cuInit` if it's provided via the `CUDA_INJECTION64_PATH` environment variable.
```

Comment:
This should be wrapped to a reasonable length like the other lines for viewing in a terminal